util/patricia-trie-ethereum/src/rlp_node_codec.rs

```diff
@@ -18,21 +18,21 @@
 
 use elastic_array::{ElasticArray1024, ElasticArray128};
 use ethereum_types::H256;
-use hashdb::Hasher;
-use keccak_hasher::KeccakHasher;
 use rlp::{DecoderError, RlpStream, Rlp, Prototype};
 use std::marker::PhantomData;
 use trie::{NibbleSlice, NodeCodec, node::Node, ChildReference};
 
 /// Concrete implementation of a `NodeCodec` with Rlp encoding, generic over the `Hasher`
 #[derive(Default, Clone)]
-pub struct RlpNodeCodec<H: Hasher> {mark: PhantomData<H>}
+pub struct RlpNodeCodec<H> {
+	mark: PhantomData<H>
+}
 
 // NOTE: what we'd really like here is:
 // `impl<H: Hasher> NodeCodec<H> for RlpNodeCodec<H> where H::Out: Decodable`
 // but due to the current limitations of Rust const evaluation we can't
 // do `const HASHED_NULL_NODE: H::Out = H::Out( … … )`. Perhaps one day soon?
-impl NodeCodec<KeccakHasher> for RlpNodeCodec<KeccakHasher> {
+impl NodeCodec<H256> for RlpNodeCodec<H256> {
```

Comment:
`RlpNodeCodec` can is no longer tied to `KeccakHasher`

Reply:
this is a problem, because now the `HASHED_NULL_NODE` will be defined but wrong for all non-keccak hashers.

Reply:
good catch!